python_modules/libraries/dagster-components/dagster_components/blueprint.py

```diff
@@ -0,0 +1,84 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Callable, Optional, TypeVar, Union
+
+from dagster import _check as check
+from dagster._record import record
+from pydantic import BaseModel
+
+# Type variable for generic class handling
+T = TypeVar("T")
+
+# Constant for blueprint attribute name
+BLUEPRINT_CLS_ATTRIBUTE = "__blueprint_cls__"
+
+
+def blueprint(
```

Comment:
I agree with the feedback that `@blueprint` doesn't feel great since the decorated target does not become a blueprint. Thoughts on bringing verb in like `@scaffold_with` / `@can_scaffold` or some modifier to indicate that the blueprint is attached to / targets the decorated entity?